tests/spec/fs.link.spec.js

```diff
@@ -111,3 +111,88 @@ describe('fs.link', function() {
     });
   });
 });
+
+
+
+
+
+describe('fs.promises.link', function() {
+  beforeEach(util.setup);
+  afterEach(util.cleanup);
+
+  it('should be a function', function() {
+    var fs = util.fs();
+    expect(fs.promises.link).to.be.a('function');
+  });
+
+  it('should create a link to an existing file', function(done) {
+    var fs = util.fs();
+
+    fs.open('/myfile', 'w+', function(error, fd) {
```

Comment:
You're mixing callbacks and promises in the same test, which is a bit confusing.  Here's how you should write something like this:

```js
var fsPromises = util.fs().promises;
var oldStats;

return fsPromises.open('/myfile', 'w+')
  .then(fd => fsPromises.close(fd))
  .then(() => fsPromises.stat('/myfile'))
  .then(result => {
    oldStats = result;
    return fsPromises.stat('/myotherfile')
  })
  .then(result => {
    // do all your expect assertions here)
  });
```

Can you fix all your tests so they only use promises?